### PR TITLE
check iframe exist before postMessage in IntersectionObserver

### PR DESCRIPTION
**Pull request by zhouyx**
#4035

### CODE REVIEW

File: src/intersection-observer.js [hunk 1]
```diff
@@ -219,6 +219,10 @@ export class IntersectionObserver extends Observable {
    * @private
    */
   flush_() {
+    if (!this.iframe_) {
+      return;
```
Comment (zhouyx) [inline src/intersection-observer.js]: If `!this.iframe` will this `IntersectionObserver` be reused. Do we need to do some clean up here?
We can probably set `this.boundFlush = null`, `this.pendingChanges_ = null` here?

Comment (jridgewell) [inline src/intersection-observer.js]: When is this ever removed?

Comment (zhouyx) [inline src/intersection-observer.js]: I don't find where `this.boundFlush` and `this.pendingChanges` are removed. But IntersectionObserver itself can be set to null in `unLayoutCallback()`.

Comment (cramforce) [inline src/intersection-observer.js]: This wouldn't actually work. I think we should create a `destroy` function and call it in all places that set intersectionObserver to null. This function can then call `clearTimeout` to make this problem go away.

**Comment by jridgewell** [comment]
LGTM.

**Pull request merged**